c_cpp_guide.md

```diff
@@ -0,0 +1,111 @@
+# Compiling C/C++ to Ewasm
+
+First an introduction, then a basic step-by-step guide, then advanced things. Warning: the Ewasm spec and tools below are subject to change.
+
+## Introduction
+
+An Ewasm contract is a WebAssembly module with the following restrictions. The module's imports must be among the [Ewasm helper functions](https://github.com/ewasm/design/blob/master/eth_interface.md) which resemble EVM opcodes to interact with the client. The module's exports must be a `main` function which takes no arguments and returns nothing, and the `memory` of the module. The module may not use floats or other sources of non-determinism.
+
+Below are quirks to be aware of when writing Ewasm contracts in C/C++. These quirks are not limited to C/C++. Things may improve as tools and Ewasm improve.
```

Comment:
> These quirks are not limited to C/C++. Things may improve as tools and Ewasm improve.

Consider deleting this part - again, not completely germane or helpful here, and might lead the reader to worry.

Reply:
Fixed, let me know if you disagree